### PR TITLE
Don't truncate min/max for iceberg delete files

### PR DESCRIPTION
**Pull request by homar**
## Description

Previously min/max were truncated to very short numbers which may have had very bad impact on performance. 

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by homar** [comment]
I used metric FULL_METRIC_CONFIG for parquet to allow storing full paths in min/max. It was impossible to do it this way for ORC so I decided to put hard limit there.
Does it make sense to do it this way or should I also make hard limit on length for parquet to have it consistent ? 
@alexjo2144 @findepi

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java [hunk 1]
```diff
@@ -146,19 +147,21 @@ public IcebergFileWriter createPositionDeleteWriter(
...
+            Map<String, String> storageProperties,
+            Optional<DataSize> stringStatisticsLimit)
     {
         switch (fileFormat) {
             case PARQUET:
-                return createParquetWriter(outputPath, POSITION_DELETE_SCHEMA, jobConf, session, hdfsContext);
+                return createParquetWriter(FULL_METRICS_CONFIG, outputPath, POSITION_DELETE_SCHEMA, jobConf, session, hdfsContext);
             case ORC:
-                return createOrcWriter(FULL_METRICS_CONFIG, outputPath, POSITION_DELETE_SCHEMA, jobConf, session, storageProperties);
+                return createOrcWriter(FULL_METRICS_CONFIG, outputPath, POSITION_DELETE_SCHEMA, jobConf, session, storageProperties, stringStatisticsLimit);
```
Comment (alexjo2144) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java:157]: We never want to truncate the stats path, right? So could we do

`DataSize.ofBytes(Long.MAX_VALUE)` here?

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java [hunk 2]
```diff
@@ -132,9 +133,9 @@ public IcebergFileWriter createDataFileWriter(
         switch (fileFormat) {
             case PARQUET:
                 // TODO use metricsConfig
-                return createParquetWriter(outputPath, icebergSchema, jobConf, session, hdfsContext);
+                return createParquetWriter(MetricsConfig.getDefault(), outputPath, icebergSchema, jobConf, session, hdfsContext);
             case ORC:
-                return createOrcWriter(metricsConfig, outputPath, icebergSchema, jobConf, session, storageProperties);
+                return createOrcWriter(metricsConfig, outputPath, icebergSchema, jobConf, session, storageProperties, Optional.empty());
```
Comment (alexjo2144) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java:138]: Maybe cleaner to not have the stats limit be optional in `createOrcWriter` and instead get it from the session here, so we always have a value

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java [hunk 3]
```diff
@@ -132,9 +133,9 @@ public IcebergFileWriter createDataFileWriter(
         switch (fileFormat) {
             case PARQUET:
                 // TODO use metricsConfig
-                return createParquetWriter(outputPath, icebergSchema, jobConf, session, hdfsContext);
+                return createParquetWriter(MetricsConfig.getDefault(), outputPath, icebergSchema, jobConf, session, hdfsContext);
```
Comment (alexjo2144) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java:136]: Is there a GH Issue to support metrics mode for Parquet? If not, can you make one

Either way, link it here

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java (hunk 1 above)
Comment (homar) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java:157]: hmm that makes sense, I will change it

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java (hunk 2 above)
Comment (homar) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java:138]: ok!

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java (hunk 1 above)
Comment (homar) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java]: Actually it has to be Integer.MAX_VALUE

**Review by findepi — APPROVED** [review]

**Pull request merged**